src/mui/input/ReferenceInput.js

```diff
@@ -108,10 +108,8 @@ export class ReferenceInput extends Component {
     }
 
     setFilter = (filter) => {
-        if (filter !== this.params.filter) {
-            this.params.filter = this.props.filterToQuery(filter);
-            this.fetchReferenceAndOptions();
-        }
+        this.params.filter = this.props.filterToQuery(this.params.filter, filter);
```

Comment:
`this.params.filter` and `filter` are confusing. I advise you to rename the `filter` variable to `searchText`